src/org/zaproxy/zap/extension/requester/HttpPanelSender.java

```diff
@@ -0,0 +1,217 @@
+
+        } catch (final Exception e) {
+            logger.error(e.getMessage(), e);
+        }
+    }
+
+    /**
+     * Go thru each listener and offer him to take over the connection. The
+     * first observer that returns true gets exclusive rights.
+     *
+     * @param httpMessage Contains HTTP request & response.
+     * @param inSocket Encapsulates the TCP connection to the browser.
+     * @param method Provides more power to process response.
+     *
+     * @return Boolean to indicate if socket should be kept open.
+     */
+    private boolean notifyPersistentConnectionListener(HttpMessage httpMessage, Socket inSocket, ZapGetMethod method) {
+        boolean keepSocketOpen = false;
+        PersistentConnectionListener listener = null;
+        synchronized (persistentConnectionListener) {
+            for (int i = 0; i < persistentConnectionListener.size(); i++) {
+                listener = persistentConnectionListener.get(i);
+                try {
+                    if (listener.onHandshakeResponse(httpMessage, inSocket, method)) {
+                        // inform as long as one listener wishes to overtake the connection
+                        keepSocketOpen = true;
+                        break;
+                    }
+                } catch (Exception e) {
+                    logger.warn(e.getMessage(), e);
+                }
+            }
+        }
+
+        return keepSocketOpen;
+    }
+
+    protected ExtensionHistory getHistoryExtension() {
+        if (extension == null) {
+            extension = ((ExtensionHistory) Control.getSingleton()
+                    .getExtensionLoader().getExtension(ExtensionHistory.NAME));
+        }
+        return extension;
+    }
+
+    @Override
+    public void cleanup() {
+        if (delegate != null) {
+            delegate.shutdown();
+            delegate = null;
+        }
+
+        final boolean isSessionTrackingEnabled = Model.getSingleton()
+                .getOptionsParam().getConnectionParam().isHttpStateEnabled();
+        getButtonUseTrackingSessionState().setEnabled(isSessionTrackingEnabled);
+    }
+
+    private HttpSender getDelegate() {
+        if (delegate == null) {
+            delegate = new HttpSender(Model.getSingleton().getOptionsParam()
+                    .getConnectionParam(), getButtonUseTrackingSessionState()
+                    .isSelected(), HttpSender.MANUAL_REQUEST_INITIATOR);
+        }
+        return delegate;
+    }
+
+    private JToggleButton getButtonFollowRedirects() {
+        if (followRedirect == null) {
+            followRedirect = new JToggleButton(new ImageIcon(
+                    HttpPanelSender.class
+                    .getResource("/resource/icon/16/118.png"))); // Arrow
+            // turn
+            // around
+            // left
+            followRedirect.setToolTipText(Constant.messages
+                    .getString("manReq.checkBox.followRedirect"));

```

Comment:
This (and others) message is from core, should those messages be duplicated in the add-on? (Might be safer.)